scenarios/kubernetes_e2e.py

```diff
@@ -0,0 +1,239 @@
+    if os.path.isfile(e2e_image_tag_override):
+        with open(e2e_image_tag_override, 'r') as tag:
+            e2e_image_tag = tag.read()
+
+    docker_env_ignore = [
+      'GOOGLE_APPLICATION_CREDENTIALS',
+      'GOROOT',
+      'HOME',
+      'PATH',
+      'PWD',
+      'WORKSPACE'
+    ]
+
+    docker_extra_args = []
+    if (os.environ.get('JENKINS_ENABLE_DOCKER_IN_DOCKER') and
+            os.environ.get('JENKINS_ENABLE_DOCKER_IN_DOCKER').lower() == 'y'):
+        docker_extra_args.extend([
+          '-v', '/var/run/docker.sock:/var/run/docker.sock',
+          '-v', '%s:/go/src/k8s.io/kubernetes' % repo,
+          '-e', 'REPO_DIR=%s' % repo,
+          '-e', 'HOST_ARTIFACTS_DIR=%s' % artifacts
+        ])
+
+    if (os.environ.get('JENKINS_USE_LOCAL_BINARIES') and
+            os.environ.get('JENKINS_USE_LOCAL_BINARIES').lower() == 'y'):
+        docker_extra_args.extend([
+          '-v', '%s/_output":/workspace/_output:ro' % workspace
+        ])
+
+    if 'KUBE_E2E_RUNNER' in os.environ:
+        docker_extra_args.extend([
+          '--entrypoint=%s', os.environ.get('KUBE_E2E_RUNNER')
+        ])
+
+
+    docker_timeout = os.environ.get('DOCKER_TIMEOUT') or '60m'
+
+    # exec
+    container = '%s-%s' % (os.environ.get('JOB_NAME'), os.environ.get('BUILD_NUMBER'))
+
+    print 'Starting...'
+
+    cmd = ['timeout', '-s', 'KILL', docker_timeout, 'docker', 'run', '--rm',
+    # For local test, timeout not work on mac
+    # cmd = ['docker', 'run', '--rm',
+      '--name=%s' % container,
+      '-v', '%s/_artifacts":/workspace/_artifacts' % workspace,
+      '-v', '/etc/localtime:/etc/localtime:ro'
+    ]
+
+    if os.environ.get('JENKINS_GCE_SSH_PRIVATE_KEY_FILE'):
+        cmd.extend(['-v', '%s:/workspace/.ssh/google_compute_engine:ro'
+            % os.environ['JENKINS_GCE_SSH_PRIVATE_KEY_FILE']])
+
+    if os.environ.get('JENKINS_GCE_SSH_PUBLIC_KEY_FILE'):
+        cmd.extend(['-v', '%s:/workspace/.ssh/google_compute_engine.pub:ro'
+            % os.environ['JENKINS_GCE_SSH_PUBLIC_KEY_FILE']])
+
+    if os.environ.get('JENKINS_AWS_SSH_PRIVATE_KEY_FILE'):
+        cmd.extend(['-v', '%s:/workspace/.ssh/kube_aws_rsa:ro'
+            % os.environ['JENKINS_AWS_SSH_PRIVATE_KEY_FILE']])
+
+    if os.environ.get('JENKINS_AWS_SSH_PUBLIC_KEY_FILE'):
+        cmd.extend(['-v', '%s:/workspace/.ssh/kube_aws_rsa.pub:ro'
+            % os.environ['JENKINS_AWS_SSH_PUBLIC_KEY_FILE']])
+
+    if os.environ.get('JENKINS_AWS_CREDENTIALS_FILE'):
+        cmd.extend(['-v', '%s:/workspace/.aws/credentials:ro'
+            % os.environ['JENKINS_AWS_CREDENTIALS_FILE']])
+
+    if os.environ.get('GOOGLE_APPLICATION_CREDENTIALS'):
+        cmd.extend(['-v', '%s:/service-account.json:ro'
+            % os.environ['GOOGLE_APPLICATION_CREDENTIALS'],
+                    '-e', 'GOOGLE_APPLICATION_CREDENTIALS=/service-account.json'])
+
+    for key, value in os.environ.items():
```

Comment:
This may be a good time to be more explicit about what environment variables we really need rather than blindly shoving everything in there.